[
    {
        "user": "U055BSWEY7M",
        "type": "message",
        "ts": "1682643720.988399",
        "client_msg_id": "225b0061-3ee5-4b09-b20c-8f471d5211cf",
        "text": "Hi. I've followed the guide here to create the necessary certificate using the XCA tool: <https:\/\/docs.developer.betfair.com\/display\/1smk3cen4v3lu3yomq5qye0ni\/Certificate+Generation+With+XCA>. I've then uploaded it successfully to the Betfair site. When I try to login using the bflw client I get the following error `Exception: Certificates not found in directory: '&lt;path to certificates folder&gt;' (make sure .crt and .key pair or a single .pem is present)`. This is self explanatory but the guide above doesn't mention creating a `.key` file or a `.pem` file! Playing around with the XCA app I have found that I can create a `.pem` file  from either exporting the key or the certificate request. I've tried both of these but on both I get the following error: `Exception: HTTPSConnectionPool(host='<http:\/\/identitysso-cert.betfair.com|identitysso-cert.betfair.com>', port=443): Max retries exceeded with url: \/api\/certlogin (Caused by SSLError(SSLError(9, '[SSL] PEM lib (_ssl.c:3874)')))`. I've had a search through <#C4H05ML2E|issues> and there's a similar post but that's for a `4044` error. Any ideas on where I could go from here?",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g26ca51c9d5d",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/26ca51c9d5dcc56f1e2255f5ec0dcf74.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "Philip",
            "real_name": "Philip Joss",
            "display_name": "Philip Joss",
            "team": "T4G9NBD2M",
            "name": "philip.joss",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1682643720.988399",
        "reply_count": 7,
        "reply_users_count": 2,
        "latest_reply": "1682713800.011759",
        "reply_users": [
            "UBS7QANF3",
            "U055BSWEY7M"
        ],
        "replies": [
            {
                "user": "UBS7QANF3",
                "ts": "1682661146.207849"
            },
            {
                "user": "U055BSWEY7M",
                "ts": "1682700427.162769"
            },
            {
                "user": "UBS7QANF3",
                "ts": "1682700737.735869"
            },
            {
                "user": "U055BSWEY7M",
                "ts": "1682701164.336049"
            },
            {
                "user": "UBS7QANF3",
                "ts": "1682705320.711769"
            },
            {
                "user": "U055BSWEY7M",
                "ts": "1682713756.666659"
            },
            {
                "user": "UBS7QANF3",
                "ts": "1682713800.011759"
            }
        ],
        "is_locked": false,
        "subscribed": false,
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "DQI",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Hi. I've followed the guide here to create the necessary certificate using the XCA tool: "
                            },
                            {
                                "type": "link",
                                "url": "https:\/\/docs.developer.betfair.com\/display\/1smk3cen4v3lu3yomq5qye0ni\/Certificate+Generation+With+XCA"
                            },
                            {
                                "type": "text",
                                "text": ". I've then uploaded it successfully to the Betfair site. When I try to login using the bflw client I get the following error "
                            },
                            {
                                "type": "text",
                                "text": "Exception: Certificates not found in directory: '<path to certificates folder>' (make sure .crt and .key pair or a single .pem is present)",
                                "style": {
                                    "code": true
                                }
                            },
                            {
                                "type": "text",
                                "text": ". This is self explanatory but the guide above doesn't mention creating a "
                            },
                            {
                                "type": "text",
                                "text": ".key",
                                "style": {
                                    "code": true
                                }
                            },
                            {
                                "type": "text",
                                "text": " file or a "
                            },
                            {
                                "type": "text",
                                "text": ".pem",
                                "style": {
                                    "code": true
                                }
                            },
                            {
                                "type": "text",
                                "text": " file! Playing around with the XCA app I have found that I can create a "
                            },
                            {
                                "type": "text",
                                "text": ".pem",
                                "style": {
                                    "code": true
                                }
                            },
                            {
                                "type": "text",
                                "text": " file  from either exporting the key or the certificate request. I've tried both of these but on both I get the following error: "
                            },
                            {
                                "type": "text",
                                "text": "Exception: HTTPSConnectionPool(host='",
                                "style": {
                                    "code": true
                                }
                            },
                            {
                                "type": "link",
                                "url": "http:\/\/identitysso-cert.betfair.com",
                                "text": "identitysso-cert.betfair.com",
                                "style": {
                                    "code": true
                                }
                            },
                            {
                                "type": "text",
                                "text": "', port=443): Max retries exceeded with url: \/api\/certlogin (Caused by SSLError(SSLError(9, '[SSL] PEM lib (_ssl.c:3874)')))",
                                "style": {
                                    "code": true
                                }
                            },
                            {
                                "type": "text",
                                "text": ". I've had a search through "
                            },
                            {
                                "type": "channel",
                                "channel_id": "C4H05ML2E"
                            },
                            {
                                "type": "text",
                                "text": " and there's a similar post but that's for a "
                            },
                            {
                                "type": "text",
                                "text": "4044",
                                "style": {
                                    "code": true
                                }
                            },
                            {
                                "type": "text",
                                "text": " error. Any ideas on where I could go from here?"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1682661146.207849",
        "client_msg_id": "ED6B179A-E2FA-4678-835A-0305DB7B6733",
        "text": "What version of Python are you running?",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "thread_ts": "1682643720.988399",
        "parent_user_id": "U055BSWEY7M",
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "8Oq",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "What version of Python are you running?"
                            }
                        ]
                    }
                ]
            }
        ]
    }
]